pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -197,7 +202,29 @@ def execute(
             self.tracker.update(batches=1, executions=len(circuits))
             self.tracker.record()
 
-        results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        if execution_config.max_workers is None:
+            max_workers = self._max_workers
+        else:
+            max_workers = execution_config.max_workers
+
+        if max_workers is None:
```

Comment:
Does `max_workers` need to be specified to enable multiprocessing? It looks like `max_workers=None` is valid in [`ProcessPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#processpoolexecutor) and just defaults to using all processors on the machine.

Reply:
The default behavior of `None` (use # cores) in multiprocessing is altered so that `tf`, `jax` and `torch` scripts are handle as before. If a jitted circuit is passed in `execute` it might already use multithreading for loops and BLAS ops, so adding multiprocessing on top could be adverse. If we want to make multiprocessing default, we could just fetch the number of cores in `__init__` and use that as default.